astropy/cosmology/flrw.py

```diff
@@ -1151,8 +1085,10 @@ def _integral_comoving_distance_z1z2(self, z1, z2):
         d : `~astropy.units.Quantity` ['length']
           Comoving distance in Mpc between each input redshift.
         """
+        z1 = z1 if not hasattr(z1, "unit") else u.Quantity(z1, cu.redshift).value
```

Comment:
These are needed because of ``ufunc`` fun. They fail on Quantities, so must be degraded to numpy arrays.
This is more a bug fix from the redshift units PR.

Reply:
When #11893 and followup are in, I think this can be avoided.